src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+type Key = 'seenAnnouncements';
+
+type SeenAnnouncements = string[];
+
+const isSeenAnnouncements = (
+  key: Key,
+  value: any,
+): value is SeenAnnouncements => {
+  return key === 'seenAnnouncements';
+};
+
+const useLocalStorage = () => {
+  const [localStorageApi] = useState({
+    /**
+     * Returns the current value associated with the given key, or null if the given key does not exist in the list associated with the object.
+     */
+    getItem: (key: Key) => {
+      const value = JSON.parse(localStorage.getItem(key));
+      if (isSeenAnnouncements(key, value)) {
```

Comment:
I wonder: is the 'announcement' code in this hook not too specific?

Reply:
The 'announcement' code is there to be able to automaticly infer the type when using getItem. When there are multiple things stored in the localStorage, this hook will be able to tell the type based on the key your passing to getItem 🙂 

[Typeguards are used for this](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types)